src/lib.rs

```diff
@@ -17,8 +20,95 @@ use tempfile::TempDir;
+    // Call http://localhost/hpos-holochain-api/install_hosted_happ
+    // for each WrappedHeaderHash to install the hosted_happ
+    let client = reqwest::Client::new();
+    // Note: Tmp preferences
+    let preferences = Preferences {
+        max_fuel_before_invoice: 1.0,
+        max_time_before_invoice: vec![86400, 0],
+        price_compute: 1.0,
+        price_storage: 1.0,
+        price_bandwidth: 1.0,
+    };
+    for happ_id in happs {
+        info!("Installing happ-id {:?}", happ_id);
+        let body = Body {
+            happ_id: happ_id.0.to_string(),
+            preferences: preferences.clone(),
+        };
+        let response = client
+            .post("http://localhost/hpos-holochain-api/install_hosted_happ")
+            .json(&body)
+            .send()
+            .await?;
+        info!("Installed happ-id {:?}", happ_id);
+        info!("Response {:?}", response);
+    }
+    Ok(())
+}
+
+pub async fn get_enabled_hosted_happs(core_happ: Happ) -> Result<Vec<WrappedHeaderHash>> {
+    let mut app_websocket = AppWebsocket::connect(42233)
+        .await
+        .context("failed to connect to holochain's app interface")?;
+    match app_websocket.get_app_info(core_happ.id_from_config()).await {
+        Some(app_info) => {
+            let zome_call_payload = ZomeCall {
+                cell_id: app_info.cell_data[0].clone().into_id(), // This works on the assumption that the core happs has HHA in the first position of the vec
+                zome_name: ZomeName::from("hha"),
+                fn_name: FunctionName::from("get_happs"),
+                payload: ExternInput::new(SerializedBytes::default()),
+                cap: None,
+                provenance: app_info.cell_data[0]
+                    .clone()
+                    .into_id()
+                    .agent_pubkey()
+                    .to_owned(),
+            };
+            let response = app_websocket.zome_call(zome_call_payload).await?;
+            match response {
+                // This is the happs list that is returned from the hha DNA
+                // https://github.com/Holo-Host/holo-hosting-app-rsm/blob/develop/zomes/hha/src/lib.rs#L54
+                // return Vec of happ_list.happ_id
+                AppResponse::ZomeCall(r) => {
+                    info!("ZomeCall Response - Hosted happs List {:?}", r);
+                    let happ_bundles: Vec<HappBundleDetails> =
+                        rmp_serde::from_read_ref(r.into_inner().bytes())?;
+                    let happ_bundle_ids: Vec<WrappedHeaderHash> =
+                        happ_bundles.into_iter().map(|happ| happ.happ_id).collect();
+                    info!("List of happ_ids {:?}", happ_bundle_ids);
+                    return Ok(happ_bundle_ids);
+                }
+                _ => return Err(anyhow!("unexpected response: {:?}", response)),
+            }
+        }
+        None => {
+            return Err(anyhow!("HHA is not installed"));
+        }
```

Comment:
No need to use `return` at the end of a scope
```suggestion
                    Ok(happ_bundle_ids)
                }
                _ => Err(anyhow!("unexpected response: {:?}", response)),
            }
        }
        None => Err(anyhow!("HHA is not installed")),
```